### PR TITLE
RNMT-2647 Updates cordova-plugin-network-information to latest version

### PR DESCRIPTION
**Pull request by pmaroco**
## Description
This changes apply the patch cb_12325_no_connection_when_connected_to_wifi [More info](https://issues.apache.org/jira/browse/CB-12325) that were being applied while adding a platform in a cordova app made using our native shell template.

I also merged the upstream into master and created a `outsystems` branched so we can use the forked (and modified) version of this plugin in our native shell more later.

## Context

Relates https://outsystemsrd.atlassian.net/browse/RNMT-2687
Closes https://outsystemsrd.atlassian.net/browse/RNMT-2647

## Type of changes

- [] Feature (change which adds functionality)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to new feature)
- [X] Fix (change which fixes an issue)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to fix)
- [ ] Performance (change which improves performance)
    - [ ] BREAKING CHANGE (existing functionality will not work as expected due to performance improvement)
- [ ] Refactor (non-breaking change that is neither feature, fix nor performance)
- [ ] Style (non-breaking change that only affects formatting and/or white-space)

## Components affected
- [x] Android platform
- [ ] iOS platform
- [ ] JavaScript
- [ ] OutSystems

## Tests

Manually tests were performed by installing this plugin into a cordova app generated using native shell template

## Checklist

- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [ ] Tests have been created
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

### CODE REVIEW

**Review by Chuckytuh — APPROVED** [review]

File: src/android/NetworkManager.java [hunk 1]
```diff
@@ -147,6 +132,69 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
...
+    @Override
+    public void onPause(boolean multitasking) {
+        this.unregisterReceiver();
+    }
+
+    @Override
+    public void onResume(boolean multitasking) {
+        super.onResume(multitasking);
+
+        this.unregisterReceiver();
```
Comment (EiyuuZack) [inline src/android/NetworkManager.java:147]: Why unregister here when it is already done during `onPause()`?

**Review by EiyuuZack — APPROVED** [review]
Just a question

File: src/android/NetworkManager.java (hunk 1 above)
Comment (pmaroco) [reply src/android/NetworkManager.java:147]: I didn't review the old patch, I'll try to identify that with the patch's creator and I'll reopen the PR if needed

Comment (EiyuuZack) [reply src/android/NetworkManager.java:147]: Given the semantics of the `unregisterReceiver()` method I do not see any harm in having it in both lifecycle callback methods. Was just wondering if it was an oversight or overzealous implementation.

File: src/android/NetworkManager.java [hunk 2]
```diff
@@ -147,6 +132,69 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
...
+                    }
+
+                    if(TYPE_NONE.equals(connectionType)) {
+                        boolean noConnectivity = intent.getBooleanExtra(ConnectivityManager.EXTRA_NO_CONNECTIVITY, false);
+                        LOG.d(LOG_TAG, "Intent no connectivity: " + noConnectivity);
+                        if(noConnectivity) {
+                            LOG.d(LOG_TAG, "Really no connectivity");
+                        } else {
+                            LOG.d(LOG_TAG, "!!! Switching to unknown, Intent states there is a connectivity.");
+                            sendUpdate(TYPE_UNKNOWN);
```
Comment (vmfo22) [inline src/android/NetworkManager.java:187]: @pmaroco why are we doing this? Can you follow up this in the original patch? The upstream plugin doesn't apply this mechanism and it's causing an unexpected result for Android 5

**Review by vmfo22 — COMMENTED** [review]
Check my question

**Pull request merged**